tools/karpenter-convert/pkg/convert/convert.go

```diff
@@ -169,43 +164,55 @@ func dropFields(buffer bytes.Buffer) string {
 	output = strings.Replace(output, "status: {}\n", "", -1)
 	output = strings.Replace(output, "  creationTimestamp: null\n", "", -1)
 	output = strings.Replace(output, "      resources: {}\n", "", -1)
-
 	return output
 }
 
 // Convert a Provisioner into a NodePool and an AWSNodeTemplate into a NodeClass.
 // If the input is of a different kind, returns an error
-func convert(resource runtime.Object, o *Context) (runtime.Object, error) {
+func convert(resource runtime.Object, o *Context) ([]runtime.Object, error) {
 	kind := resource.GetObjectKind().GroupVersionKind().Kind
 	switch kind {
 	case "Provisioner":
-		return convertProvisioner(resource, o), nil
+		provisioner := resource.(*corev1alpha5.Provisioner)
+
+		var providerObj runtime.Object
+		var err error
+		if provider := provisioner.Spec.Provider; provider != nil {
+			providerObj, err = convertProvider(provider.Raw, provisioner.Name)
+			if err != nil {
+				return nil, fmt.Errorf("converting spec.provider for Provisioner, %w", err)
+			}
+			provisioner.Spec.ProviderRef = &corev1alpha5.MachineTemplateRef{
+				Name: provisioner.Name,
```

Comment:
should this be providerObj.Name? looks like we make the Provider the provisioner name anyway in the `convertProvider` call, but would make more sense if we just use the providerObj name.

Reply:
Sure, makes sense